docs/Android.md

```diff
@@ -94,27 +95,18 @@ Create a simple Swift file named `hello.swift`:
 print("Hello, Android")
 ```
 
-To compile it, we need to make sure the correct linker is used. Symlink the
-gold linker in the Android NDK into your `PATH`:
-
-```
-$ sudo ln -s \
-    /path/to/android-ndk-r16/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/arm-linux-androideabi/bin/ld.gold \
-    /usr/bin/armv7-none-linux-androideabi-ld.gold
-```
-
 Then use the built Swift compiler from the previous step to compile a Swift
 source file, targeting Android:
 
 ```
-$ NDK_PATH="path/to/android-ndk16"
-$ build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swiftc \                      # The Swift compiler built in the previous step.
-                                                                                 # The location of the tools used to build Android binaries
-    -tools-directory ${NDK_PATH}/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/arm-linux-androideabi/bin
-    -target armv7-none-linux-androideabi \                                       # Targeting android-armv7.
-    -sdk ${NDK_PATH}/platforms/android-21/arch-arm \                               # Use the same NDK path and API version as you used to build the stdlib in the previous step.
-    -L ${NDK_PATH}/sources/cxx-stl/llvm-libc++/libs/armeabi-v7a \                  # Link the Android NDK's libc++ and libgcc.
-    -L ${NDK_PATH}/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/lib/gcc/arm-linux-androideabi/4.9.x \
+$ NDK_PATH="path/to/android-ndk21"
+$ build/Ninja-ReleaseAssert/swift-linux-x86_64/bin/swiftc \     # The Swift compiler built in the previous step.
+                                                                # The location of the tools used to build Android binaries
+    -tools-directory ${NDK_PATH}/toolchains/llvm/prebuilt/linux-x86_64/bin/ \
+    -target armv7a-none-linux-androideabi \                     # Targeting android-armv7.
```

Comment:
This really should be `armv7-unknown-linux-androideabi -march=armv7-a`.

Reply:
This is a `swiftc` flag to build the Swift example and invoke the `clang++` linker driver, I don't believe there's a `-march` flag accepted by `swiftc`. As for armv7 vs armv7a, [Jake Petroules asked that I use the latter before because it's the official triple](https://github.com/apple/swift-package-manager/pull/2396#discussion_r342175073), and `none` vs `unknown` for the vendor are both ignored. I honestly don't care what variation of this target triple is used, so just let me know.

Reply:
Ping, this is the last remaining issue for this pull, just let me know what you prefer given the details I linked and mentioned.

Reply:
Finally built the Android-armv7 SDK by using these commands and as expected, the above flags didn't work:
```
<unknown>:0: error: unknown argument: '-march=armv7-a'
```
Instead, the flag I used worked, as did `-target armv7-unknown-linux-androideabi -Xclang-linker -march=armv7-a`, tested by building the `hello world` executable both ways and running it.

Since both ways work once your suggestion is properly modified, I don't really care what's used, just let me know.

Reply:
I experimented with this a bit more and if I cross-compiled the Swift object file first, to avoid any transient strings from each compiler invocation, then linked it into an executable by using this command with either of these flag combos or a couple more I tried, they all produce identical executables with the same hashes, as the final linker commands run are identical.

Since the suggestion makes no difference, the flag used now should be fine.